.github/workflows/build.yml

```diff
@@ -0,0 +1,51 @@
+# General workflow to build Metal
+
+name: Metal build
+
+on: push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        distribution: [ temurin, corretto ]
+        java-version: [ 11, 17 ]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # Fetch all history for Sonar analysis
+          fetch-depth: 0
+
+      - name: Set up Cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ matrix.java-version }}
+          distribution: ${{ matrix.distribution }}
+
+      - name: Build with Maven and analyze with Sonar
+        # Analyze only the Temurin 11 build
+        if: matrix.distribution == 'temurin' && matrix.java-version == '11'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: mvn -B --update-snapshots -Dmaven.test.failure.ignore=true package org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.projectKey=parsingdata_metal
```

Comment:
As discussed offline: maybe add a comment to explain that `parsingdata_metal` projectKey is generated by the SonarCloud GitHub App during install of the app? This to avoid that this key is renamed. The value is not found anywhere else, so one may assume that this key is defined here and that it can be adjusted.